dev-java/protobuf-java/protobuf-java-4.28.0.ebuild

```diff
@@ -0,0 +1,163 @@
+# Copyright 2008-2024 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=8
+
+JAVA_PKG_IUSE="doc source test"
+MAVEN_ID="com.google.protobuf:protobuf-java:${PV}"
+JAVA_TESTING_FRAMEWORKS="junit-4"
+
+inherit java-pkg-2 java-pkg-simple cmake
+
+DESCRIPTION="Core Protocol Buffers library"
+HOMEPAGE="https://protobuf.dev"
+# Currently we bundle the binary version of truth.jar used only for tests, we don't install it.
+# And we build artifact 4.28.0 from the 28.0 tarball in order to allow sharing the tarball with
+# dev-libs/protobuf.
+MY_PV4="${PV#4.}"
+MY_PV="${MY_PV4/_rc/-rc}"
+MY_P="protobuf-${MY_PV}.tar.gz"
+TV="1.1.3"
+SRC_URI="https://github.com/protocolbuffers/protobuf/releases/download/v${MY_PV}/${MY_P}
+	test? ( https://repo1.maven.org/maven2/com/google/truth/truth/${TV}/truth-${TV}.jar )"
+S="${WORKDIR}/protobuf-${MY_PV}"
+
+LICENSE="BSD"
+SLOT="0"
+KEYWORDS="~amd64 ~arm ~arm64 ~ppc64 ~x86 ~amd64-linux ~x86-linux ~x64-macos"
+IUSE="system-protoc"
+
+BDEPEND="
+	system-protoc? ( ~dev-libs/protobuf-${MY_PV4}:0[protoc] )
+	!system-protoc? ( >=dev-cpp/abseil-cpp-20230802.0 )
+"
+DEPEND="
+	>=virtual/jdk-1.8:*
+	test? (
+		dev-java/guava:0
+		dev-java/mockito:4
+	)
+"
+RDEPEND=">=virtual/jre-1.8:*"
+
+JAVA_AUTOMATIC_MODULE_NAME="com.google.protobuf"
+JAVA_JAR_FILENAME="protobuf.jar"
+JAVA_RESOURCE_DIRS="java/core/src/main/resources"
+JAVA_SRC_DIR="java/core/src/main/java"
+
+JAVA_TEST_GENTOO_CLASSPATH="guava,junit-4,mockito-4"
+JAVA_TEST_SRC_DIR="java/core/src/test/java"
+
+run-protoc() {
+	if use system-protoc; then
+		protoc $1
+	else
+		"${BUILD_DIR}"/protoc $1
+	fi
+}
+
+src_prepare() {
+	# If the corrsponding version of system-protoc is not available we build protoc locally
+	if use system-protoc; then
+		:
+	else
+		cmake_src_prepare
+	fi
```

Comment:
```suggestion
	if ! use system-protoc; then
		cmake_src_prepare
	fi
```

Reply:
would have been glad to getting such proposal when starting with touching this package first.  will keep it for next  version in some few or more days.